src/ansys/hps/client/client.py

```diff
@@ -165,6 +168,31 @@ def __init__(
                 requests.packages.urllib3.exceptions.InsecureRequestWarning
             )
 
+        self.auth_url = auth_url
+        self.auth_api_url = (auth_url or url) + f"/auth/"
+
+        if not auth_url:
+            with requests.session() as session:
+                session.verify = self.verify
+                jms_info_url = url.rstrip("/") + "/jms/api/v1"
+                resp = session.get(jms_info_url)
+                if resp.status_code != 200:
+                    raise RuntimeError(
+                        f"Failed to contact jms info endpoint {jms_info_url}, \
+                            status code {resp.status_code}: {resp.content.decode()}"
+                    )
+                else:
+                    jms_data = resp.json()
+                    if "services" in jms_data and "external_auth_url" in jms_data["services"]:
+                        self.auth_url = resp.json()["services"]["external_auth_url"]
+                    else:
+                        log.warning(
+                            "Legacy JMS service does not include external_auth_url. \
+                                Generating auth_url..."
+                        )
+                        if realm and not realm.isspace() and "auth/realms" not in url:
+                            self.auth_url = f"{url.rstrip('/')}/auth/realms/{realm}"
+
```

Comment:
can we move this block into a separate function?

Reply:
also, why do we need to check for `"auth/realms" not in url`, when can that happen?

Reply:
Honestly, I don't have much idea of how this could be used.  If you aren't worried, I can not check.